### PR TITLE
feat: add improved prune() function

### PR DESCRIPTION
**Pull request by Duologic**
It often comes up that `std.prune()` is recursive, although the documentation states that it is, this still catches people off guard.

The function introduced here provides options to disable recursion and only prune `null` values.

### CODE REVIEW

File: reflect.libsonnet [hunk 1]
```diff
@@ -0,0 +1,47 @@
+local d = import 'github.com/jsonnet-libs/docsonnet/doc-util/main.libsonnet';
+
+{
+  '#': d.pkg(
+    name='reflect',
+    url='github.com/jsonnet-libs/xtd/reflect.libsonnet',
+    help='`reflect` implements helper functions for processing json data.',
```
Comment (jdbaldry) [inline reflect.libsonnet:7]: nit: 
```suggestion
    help='`reflect` implements helper functions for processing JSON data.',
```

File: reflect.libsonnet [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      else if only_null
+      then true
+      else if std.isArray(b)
+      then std.length(b) > 0
+      else if std.isObject(b)
+      then std.length(b) > 0
+      else true;
+    local doRecursion(x) =
+      if recurse
+      then self.prune(x, only_null=only_null)
```
Comment (jdbaldry) [inline reflect.libsonnet:31]: nit: I think it's more common to see that the recursive call uses the exact same arguments as the initial call because it explicitly demonstrates that the recursive call is gonna continue the same behavior.

```suggestion
      then self.prune(x, recurse, only_null)
```

File: test/reflect_test.jsonnet
Comment (jdbaldry) [inline test/reflect_test.jsonnet:1]: For completeness, I think that all primitives need to exist in the test objects or at least zero-length arrays to demonstrate that the same `std.prune` behavior is preserved.
